File: php/test/Behavioral/MediatorTest.php
<?

require_once 'src/Behavioral/Mediator/DvdMediator.php';
require_once 'src/Behavioral/Mediator/DvdLowercaseTitle.php';
require_once 'src/Behavioral/Mediator/DvdUppercaseTitle.php';

class Behavioral_MediatorTest extends PHPUnit_Framework_TestCase
{
   public function test()
   {
		$dvdMediator = new DvdMediator();
		$dvdLower = new DvdLowercaseTitle("Mulholland Dr.", $dvdMediator);
		$dvdUp = new DvdUppercaseTitle($dvdLower, $dvdMediator);

		$this->assertEquals($dvdLower->lowercasetitle ,'mulholland dr.',"Lowercase LC title :" . $dvdLower->lowercasetitle);
		$this->assertEquals( $dvdLower->getTitle() ,'Mulholland Dr.',"Lowercase super title :");
		$this->assertEquals( $dvdUp->uppercasetitle ,'MULHOLLAND DR.',"Upcase UC title :" );
		$this->assertEquals( $dvdUp->getTitle() , 'Mulholland Dr.',"Upcase super title :");

		$dvdLower->setSuperTitleLowercase();

		$this->assertEquals( $dvdLower->lowercasetitle ,'mulholland dr.',"Lowercase LC title :");
		$this->assertEquals( $dvdLower->getTitle() ,'mulholland dr.',"Lowercase super title :");
		$this->assertEquals( $dvdUp->uppercasetitle ,'MULHOLLAND DR.',"Upcase UC title :");
		$this->assertEquals( $dvdUp->getTitle() ,'Mulholland Dr.',"Upcase super title :");

	}
}

File: clojure/test/patterns/behavioral/mediator_test.clj
(ns patterns.behavioral.mediator-test
  (:require
      [clojure.test :refer :all]
      [patterns.behavioral.mediator :refer :all]
   ))

	(let
		[title "Mulholland Dr."
		 dvdLower    (dvdlowercasetitle (ref title))
		 dvdUpper    (dvduppercasetitle (ref title))
		 mediator (dvdmediator :lower (ref dvdLower) :upper (ref dvdUpper))]

	(deftest mediator-test

		(testing "Upper and Lower maps have same title"
			(is (= (deref (:title dvdUpper)) (deref (:title dvdLower)))))

		(testing "Lower has its own lowercased title"
			(is (= "mulholland dr." (:lowercasetitle dvdLower))))

		(testing "Upper has its own uppercased title"
			(is (= "MULHOLLAND DR." (:uppercasetitle dvdUpper))))
	
		(set-super-lowercase-title mediator)

		(testing "Lower has its own lowercased title, still"
			(is (= "mulholland dr." (:lowercasetitle dvdLower))))

		(testing "Shared title is now lowercased, in lower"
			(is (= "mulholland dr." (deref (:title dvdLower)))))
		(testing "Shared title is now lowercased, in upper"
			(is (= "mulholland dr." (deref (:title dvdUpper)))))

		(testing "Upper's uppercasetitle is not affected"
			(is (= "MULHOLLAND DR." (:uppercasetitle dvdUpper))))
	))


